src/index.ml

```diff
@@ -656,45 +656,25 @@ struct
       incr n;
       !n
 
-  let merge' ?(blocking = false) ?filter ?(hook = fun _ -> ()) ~witness
-      ?(force = false) t =
-    let yield () = check_pending_cancel t in
-    let counter = Mtime_clock.counter () in
-    let merge_id = merge_counter () in
-    let msg = Fmt.strf "merge { id=%d }" merge_id in
-    Semaphore.acquire msg t.merge_lock;
-    let merge_lock_wait = Mtime_clock.count counter in
-    Log.info (fun l ->
-        let pp_forced ppf () = if force then Fmt.string ppf "; force=true" in
-        l "[%s] merge started { id=%d%a }" (Filename.basename t.root) merge_id
-          pp_forced ());
-    Stats.incr_nb_merge ();
-    let log_async =
-      let io =
-        let log_async_path = Layout.log_async ~root:t.root in
-        IO.v ~flush_callback:t.config.flush_callback ~fresh:true
-          ~generation:(Int63.succ t.generation) ~fan_size:Int63.zero
-          log_async_path
-      in
-      let mem = Tbl.create 0 in
-      { io; mem }
+  (** Merges the entries in [t.log] into the data file, ensuring that concurrent
+      writes are not lost.
+
+      The caller must ensure the following:
+
+      - [t.log] has been loaded;
+      - [t.log_async] has been created;
+      - [t.merge_lock] is acquired before entry, and released immediately after
+        this function returns or raises an exception. *)
+  let unsafe_perform_merge =
+    let compare_entry (e : Entry.t) (e' : Entry.t) =
```

Comment:
you can probably move this out of the function (and use `Int.compare` to make sure we don't use the polymorphic compare function) and move the function argument at the "right" place

Reply:
Done in the latest commit.